op-interop-mon/monitor/metric_collector.go

```diff
@@ -72,46 +72,103 @@ func (m *MetricCollector) CollectMetrics() {
+		// Initialize executing ranges for this chain if not seen before
+		if _, exists := executingRanges[job.executingChain]; !exists {
+			executingRanges[job.executingChain] = struct {
+				min, max uint64
+			}{
+				min: job.executingBlock.Number,
+				max: job.executingBlock.Number,
+			}
+		}
+
+		// Initialize initiating ranges for this chain if not seen before
+		if _, exists := initiatingRanges[job.initiating.ChainID]; !exists {
+			initiatingRanges[job.initiating.ChainID] = struct {
+				min, max uint64
+			}{
+				min: job.initiating.BlockNumber,
+				max: job.initiating.BlockNumber,
+			}
+		}
+
+		// Update executing ranges
+		execRange := executingRanges[job.executingChain]
+		if job.executingBlock.Number < execRange.min {
+			execRange.min = job.executingBlock.Number
+		}
+		if job.executingBlock.Number > execRange.max {
+			execRange.max = job.executingBlock.Number
+		}
+		executingRanges[job.executingChain] = execRange
+
+		// Update initiating ranges
+		initRange := initiatingRanges[job.initiating.ChainID]
+		if job.initiating.BlockNumber < initRange.min {
+			initRange.min = job.initiating.BlockNumber
+		}
+		if job.initiating.BlockNumber > initRange.max {
+			initRange.max = job.initiating.BlockNumber
+		}
+		initiatingRanges[job.initiating.ChainID] = initRange
+
 		statuses := job.Statuses()
 		if len(statuses) == 0 {
 			m.log.Warn("Job has no statuses", "job", job)
 			continue
 		}
 		current := statuses[len(statuses)-1].String()
-		// Lazy increment the executing message metrics
-		if _, ok := executingMessages[job.executingChain]; !ok {
-			executingMessages[job.executingChain] = make(map[uint64]map[common.Hash]map[string]int)
-		}
-		if _, ok := executingMessages[job.executingChain][job.executingBlock.Number]; !ok {
-			executingMessages[job.executingChain][job.executingBlock.Number] = make(map[common.Hash]map[string]int)
-		}
-		if _, ok := executingMessages[job.executingChain][job.executingBlock.Number][job.executingBlock.Hash]; !ok {
-			executingMessages[job.executingChain][job.executingBlock.Number][job.executingBlock.Hash] = make(map[string]int)
+
+		// Log invalid statuses
+		if current == jobStatusInvalid.String() {
+			m.log.Warn("Invalid Executing Message Detected",
+				"executing_chain_id", job.executingChain,
+				"initiating_chain_id", job.initiating.ChainID,
+				"executing_block_height", job.executingBlock.Number,
+				"initiating_block_height", job.initiating.BlockNumber,
+				"executing_block_hash", job.executingBlock.Hash,
+			)
 		}
-		if _, ok := executingMessages[job.executingChain][job.executingBlock.Number][job.executingBlock.Hash][current]; !ok {
-			executingMessages[job.executingChain][job.executingBlock.Number][job.executingBlock.Hash][current] = 0
+
+		// Check for multiple initiating hashes
+		initiatingHashes := job.InitiatingHashes()
+		if len(initiatingHashes) > 1 {
+			m.log.Warn("Initiating BlockNumber found multiple Blocks (potential reorg of initiating block)",
```

Comment:
```suggestion
			m.log.Warn("Initiating BlockNumber found multiple Blocks (reorg of initiating block)",
```
Are we sure enough to say this? I feel like we log "maybe a reorg" in op-node a lot and it can be confusing.

Reply:
If you pull a block 5 from Chain A and get 0x123,

and then later pull block 5 from Chain A and get 0xABC,

*some* form of reorg has occurred, this is basically the textbook example. Any explanation for having multiple blocks at the same height indicate some sort of reorg, whether it be the node corrupting and fixing itself (extremely rare), and L2 equivocation, or an L1 reorg, the chain at the given height has changed.

I know the "L1 head indicates possible reorg" sort of log you're referencing in `op-node`, and I agree that one is confusing. But this one is pretty clear cut IMO.

Have committed your version, thanks :)